php/src/php/Properties.cpp

```diff
@@ -493,9 +497,17 @@ handleFreeStorage(zend_object* object)
 extern "C"
 #endif
 static zend_object*
+#if PHP_VERSION_ID >= 80000
+handleClone(zend_object* zobj)
+#else
 handleClone(zval* zv)
+#endif
 {
+#if PHP_VERSION_ID >= 80000
+    Ice::PropertiesPtr p = *Wrapper<Ice::PropertiesPtr>::fetch(zobj)->ptr;
```

Comment:
Shouldn't we check ptr is set before dereferencing it?

Reply:
This ptr is never null

Reply:
The ptr is initialized to 0 so it can be 0 after initialization, in ::value implementation with is somehow similar we do check for null ptr

Reply:
We have an assert right after to check that it's not null. You can't clone a null value.

Reply:
I can split it, and we can just assert on the ptr before dereferencing it, but I don't really see the point

Reply:
I was confused because we use `value` below that does null checking, maybe the other branch should just use `extract` and dereference the pointer like this one? anyway, fine to keep it as is.

The assertion that the ptr is set is unrelated to this, I was worried that the pointer to the ptr was not.